test-tool/test_receive_copy_results_copy_status.c

```diff
@@ -35,10 +35,22 @@ test_receive_copy_results_copy_status(void)
         int  offset = XCOPY_DESC_OFFSET, list_id = 1;
         struct iscsi_data data;
         unsigned char *xcopybuf;
+        struct scsi_command_descriptor *cd;
 
         logging(LOG_VERBOSE, LOG_BLANK_LINE);
         logging(LOG_VERBOSE, "Test RECEIVE COPY RESULTS, COPY STATUS");
 
+        logging(LOG_VERBOSE, "Check that RECEIVE COPY RESULTS is supported.");
+        cd = get_command_descriptor(SCSI_OPCODE_RECEIVE_COPY_RESULTS,
```

Comment:
Implementing REPORT SUPPORTED OPERATION CODES is not mandatory. E.g. LIO does not implement it. I think we need a better solution to test whether or not RECEIVE COPY RESULTS is supported.

Reply:
@bvanassche What do you think if we try RSOC first, then try to send the command and if neither works the tests can be skipped?

BTW, we've got RSOC implemented for TCM just haven't managed to send patches yet.

Reply:
Another option could be Third-party Copy VPD (0x8f) page as defined in SPC:
```
/* 5.19.1 General considerations for third-party copies */
Application clients and other copy managers request third-party copy operations by sending commands (see 5.19.3) to a copy manager for processing by that copy manager. Copy managers provide information about their capabilities to application clients in the Third-party Copy VPD page (see 7.7.18).
/* 7.7.18.4.1 Supported Commands third-party copy descriptor overview */
The Supported Commands third-party copy descriptor (see table 587) indicates which commands (i.e., combi- nations of operation code and service action) (see 5.19.3) the copy manager supports.
```

Unfortunately it's not implemented in TCM, but it's something that might help feature discovery or test selection along with RSOC.

Reply:
Yes, implement Third-Party Copy VPD in LIO is a good idea. We will do it.